todoapp/app/src/androidTest/java/com/example/android/architecture/blueprints/todoapp/data/source/local/TasksDaoTest.kt

```diff
@@ -0,0 +1,172 @@
+
+    private lateinit var database: ToDoDatabase
+
+    @Before fun initDb() {
+        // using an in-memory database because the information stored here disappears when the
+        // process is killed
+        database = Room.inMemoryDatabaseBuilder(InstrumentationRegistry.getContext(),
+                ToDoDatabase::class.java).build()
+    }
+
+    @After fun closeDb() {
+        database.close()
+    }
+
+    @Test fun insertTaskAndGetById() {
+        // When inserting a task
+        database.taskDao().insertTask(TASK)
+
+        // When getting the task by id from the database
+        val loaded = database.taskDao().getTaskById(TASK.id)
+
+        // The loaded data contains the expected values
+        assertTask(loaded, "id", "title", "description", true)
+    }
+
+    @Test fun insertTaskReplacesOnConflict() {
+        //Given that a task is inserted
+        database.taskDao().insertTask(TASK)
+
+        // When a task with the same id is inserted
+        val newTask = Task("title2", "description2", "id").apply {
+            isCompleted = true
+        }
+        database.taskDao().insertTask(newTask)
+
+        // When getting the task by id from the database
+        val loaded = database.taskDao().getTaskById(TASK.id)
+
+        // The loaded data contains the expected values
+        assertTask(loaded, "id", "title2", "description2", true)
+    }
+
+    @Test fun insertTaskAndGetTasks() {
+        // When inserting a task
+        database.taskDao().insertTask(TASK)
+
+        // When getting the tasks from the database
+        val tasks = database.taskDao().getTasks()
+
+        // There is only 1 task in the database
+        assertThat(tasks.size, `is`(1))
+        // The loaded data contains the expected values
+        assertTask(tasks[0], "id", "title", "description", true)
+    }
+
+    @Test fun updateTaskAndGetById() {
+        // When inserting a task
+        database.taskDao().insertTask(TASK)
+
+        // When the task is updated
+        val updatedTask = Task("title2", "description2", "id").apply {
+            isCompleted = true
+        }
+        database.taskDao().updateTask(updatedTask)
+
+        // When getting the task by id from the database
+        val loaded = database.taskDao().getTaskById("id")
+
+        // The loaded data contains the expected values
+        assertTask(loaded, "id", "title2", "description2", true)
+    }
+
+    @Test fun updateCompletedAndGetById() {
+        // When inserting a task
+        database.taskDao().insertTask(TASK)
+
+        // When the task is updated
+        database.taskDao().updateCompleted(TASK.id, false)
+
+        // When getting the task by id from the database
+        val loaded = database.taskDao().getTaskById("id")
+
+        // The loaded data contains the expected values
+        assertTask(loaded, TASK.id, TASK.title, TASK.description, false)
+    }
+
+    @Test fun deleteTaskByIdAndGettingTasks() {
+        //Given a task inserted
```

Comment:
introduce a space between `//` and text throughout

Reply:
Done (cherry-picked dev-todo-mvp-kotlin modifications).